File: app/Models/Temporada.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Temporada extends Model
{
    use HasFactory, SoftDeletes;

    protected $table = 'temporadas';
    protected $primaryKey = 'id_temporada';
    public $incrementing = true;
    protected $keyType = 'int';

    protected $fillable = [
        'nombre',
        'fecha_inicio',
        'fecha_fin',
        'modificador_precio',
    ];
}

File: src/java/models/Temporada.java
package models;

import models.enums.NombreTemporada;
import java.time.LocalDate;

public class Temporada extends ClaseBase {
    private int idTemporada;
    
    private NombreTemporada nombre;
    
    private LocalDate fechaInicio;
    
    private LocalDate fechaFin;
    
    private double modificadorPrecio;

    public int getIdTemporada() {
        return idTemporada;
    }

    public void setIdTemporada(int idTemporada) {
        this.idTemporada = idTemporada;
    }

    public NombreTemporada getNombre() {
        return nombre;
    }

    public void setNombre(NombreTemporada nombre) {
        this.nombre = nombre;
    }

    public LocalDate getFechaInicio() {
        return fechaInicio;
    }

    public void setFechaInicio(LocalDate fechaInicio) {
        this.fechaInicio = fechaInicio;
    }

    public LocalDate getFechaFin() {
        return fechaFin;
    }

    public void setFechaFin(LocalDate fechaFin) {
        this.fechaFin = fechaFin;
    }

    public double getModificadorPrecio() {
        return modificadorPrecio;
    }

    public void setModificadorPrecio(double modificadorPrecio) {
        this.modificadorPrecio = modificadorPrecio;
    }
}